plugins/module_utils/network/nxos/rm_templates/ospfv2.py

```diff
@@ -0,0 +1,1099 @@
+        if include_stub:
+            command += " include-stub"
+        if on_startup:
+            command += " on-startup"
+            if on_startup.get("wait_period"):
+                command += " {wait_period}".format(**on_startup)
+            if on_startup.get("wait_for_bgp_asn"):
+                command += " wait-for bgp {wait_for_bgp_asn}".format(
+                    **on_startup
+                )
+        if summary_lsa:
+            command += " summary-lsa"
+            if summary_lsa.get("max_metric_value"):
+                command += " {max_metric_value}".format(**summary_lsa)
+
+    return command
+
+
+def _tmplt_area_nssa(area):
+    nssa = area["nssa"]
+    command = "area {area_id} nssa".format(**area)
+    if nssa.get("set") is False:
+        command = "no {0}".format(command)
+    else:
+        for attrib in [
+            "no_summary",
+            "no_redistribution",
+            "default_information_originate",
+        ]:
+            if nssa.get(attrib):
+                command += " {0}".format(attrib.replace("_", "-"))
+    return command
+
+
+def _tmplt_area_nssa_translate(area):
+    translate = area["nssa"]["translate"]["type7"]
+    command = "area {area_id} nssa translate type7".format(**area)
+    for attrib in ["always", "never", "supress_fa"]:
+        if translate.get(attrib):
+            command += " {0}".format(attrib.replace("_", "-"))
+    return command
+
+
+def _tmplt_area_ranges(arange):
+    command = "area {area_id} range {prefix}".format(**arange)
+    if arange.get("not_advertise") is True:
+        command += " not-advertise"
+    if "cost" in arange:
+        command += " cost {cost}".format(**arange)
+    return command
+
+
+def _tmplt_area_ranges(arange):
+    command = "area {area_id} range {prefix}".format(**arange)
+    if arange.get("not_advertise") is True:
+        command += " not-advertise"
+    if "cost" in arange:
+        command += " cost {cost}".format(**arange)
+    return command
+
+
+def _tmplt_summary_address(proc):
+    command = "summary-address {prefix}".format(**proc)
+    if proc.get("tag"):
+        command += " tag {tag}".format(**proc)
+    elif proc.get("not_advertise"):
+        command += " not-advertise"
+    return command
+
+
+def _tmplt_area_stub(area):
+    stub = area["stub"]
+    command = "area {area_id} stub".format(**area)
+    if stub.get("set") is False:
+        command = "no {0}".format(command)
+    elif stub.get("no_summary"):
+        command += " no-summary"
+    return command
+
+
+def _tmplt_redistribute(redis):
+    command = "redistribute {protocol}".format(**redis)
+    if redis.get("id"):
+        command += " {id}".format(**redis)
+    if redis.get("route_map"):
+        command += " route-map {route_map}".format(**redis)
+    return command
+
+
+def _tmplt_capability_vrf_lite(proc):
+    command = "capability vrf-lite"
+    vrf_lite = proc["capability"]["vrf_lite"]
+    if vrf_lite.get("set") is False:
```

Comment:
is it required to explicitly check for False

Reply:
Yes. Consider a scenario where "capability vrf-lite evpn" is not configured on the box and we run a task that looks has:
```yaml
- process_id: 100
   vrfs:
    - vrf: blue
      capability:
        vrf_lite:
          evpn: True
```

This should push "capability vrf-lite evpn", however, if the explicit check for False is in not given, it would instead send "no capability vrf-lite". This is because the dictionary that reaches this method will be something like `'capability': {'vrf_lite': {'evpn': True}}`, i.e., missing the "set" key and the method would fail to differentiate between absence of "set"
"set: False".

In any case, I feel having these explicit checks make the code easier to read and maintain.

Valid for all the other review comments regarding the same.